ros2bag/ros2bag/api/__init__.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from argparse import ArgumentTypeError
+import os
+from typing import Dict
+from typing import Optional
+
+from rclpy.duration import Duration
+from rclpy.qos import QoSDurabilityPolicy
+from rclpy.qos import QoSHistoryPolicy
+from rclpy.qos import QoSLivelinessPolicy
+from rclpy.qos import QoSProfile
+from rclpy.qos import QoSReliabilityPolicy
+
+# This map needs to be updated when new policies are introduced
+POLICY_MAP = {
+    'history': QoSHistoryPolicy.get_from_short_key,
+    'reliability': QoSReliabilityPolicy.get_from_short_key,
+    'durability': QoSDurabilityPolicy.get_from_short_key,
+    'liveliness': QoSLivelinessPolicy.get_from_short_key
+}
+
+
+def is_dict_valid_duration(duration_dict: Dict[str, int]) -> bool:
+    """Check if dictionary contains the keys `sec` and `nsec`."""
+    return all(key in duration_dict for key in ['sec', 'nsec'])
+
+
+def dict_to_duration(time_dict: Optional[Dict[str, int]]) -> Duration:
+    """Convert a QoS duration profile from YAML into an rclpy Duration."""
+    if time_dict:
+        if is_dict_valid_duration(time_dict):
+            return Duration(seconds=time_dict.get('sec'), nanoseconds=time_dict.get('nsec'))
+        else:
+            raise ValueError(
+                'Time overrides must include both seconds (sec) and nanoseconds (nsec).')
+    else:
+        return Duration()
+
+
+def validate_qos_profile_overrides(qos_profile_dict: Dict) -> Dict[str, Dict]:
```

Comment:
This name is misleading. I would expect "validate" to return a bool, and the dict logic is not really necessary. Let's have this do the minimal useful functionality for a single item.

```
def interpret_yaml_as_qos_profile(qos_profile_dict: dict) -> rclpy.qos.QoSProfile:
```

The users of this function can do name: profile mapping themselves

```
qos_profile_overrides = {
  key: interpret_yaml_as_qos_profile(value) for key, value in qos_profile_dict
}
```